lib/trino-record-decoder/src/main/java/io/trino/decoder/avro/AvroColumnDecoder.java

```diff
@@ -127,7 +128,18 @@ private boolean isSupportedPrimitive(Type type)
 
     public FieldValueProvider decodeField(GenericRecord avroRecord)
     {
-        Object avroColumnValue = locateNode(avroRecord, columnMapping);
+        Object avroColumnValue;
+        try {
+            avroColumnValue = locateNode(avroRecord, columnMapping);
+        }
+        catch (AvroRuntimeException e) {
```

Comment:
when does this happen?

Reply:
when the referenced field does not exist (previously it would return null, not avro library throws)

Reply:
So ideally we would hit this only in case where we provide a manual column mapping (via the `mapping` field in the topic description) and not when the schema is being fetched from registry? Or am I missing some case? Because when schema from registry is used we know that the fields present in the message MUST also exist in the correct version of schema.

Reply:
Yes. Or the schema registry is wrong (but I don't know whether it's actually possible).